cpp/docs/DEVELOPER_GUIDE.md

```diff
@@ -263,11 +268,14 @@ a `string_scalar` holds a single string. The data for the stored value resides i
 |timestamp|`timestamp_scalar<T>` | `T` can be `timestamp_D`, `timestamp_s`, etc.|
 |duration|`duration_scalar<T>` | `T` can be `duration_D`, `duration_s`, etc.|
 |string|`string_scalar`| This class object is immutable|
+|list|`list_scalar`| Underlying data can be any type supported by cudf |
 
 ### Construction
 `scalar`s can be created using either their respective constructors or using factory functions like 
 `make_numeric_scalar()`, `make_timestamp_scalar()` or `make_string_scalar()`. 
 
+// TODO: add details for `list_scalar`
+
```

Comment:
Finish these TODOs before merging?

Reply:
This PR did not introduce any new factory functions for `list_scalar`, aka `make_list_scalar(...)`. I'm not quite sure at this point what should the arguments be for this.

Reply:
So do we need these TODOs?

Reply:
The paragraph above suggests a factory function is available for each type of scalars. Maybe instead of adding `TODO` we should rephrase the words above to exclude `list_scalar`.